src/mqtt/packet/mqtt_string.rs

```diff
@@ -681,6 +681,59 @@ impl TryFrom<String> for MqttString {
     }
 }
 
+/// Trait for types that can be converted into `MqttString`
+///
+/// This trait provides a flexible way to create `MqttString` instances from various string types.
+/// It allows you to pass pre-constructed `MqttString` objects directly without additional
+/// heap allocations, which is particularly useful in multi-threaded scenarios.
+///
+/// # Examples
+///
+/// ```ignore
+/// use mqtt_protocol_core::mqtt::packet::{IntoMqttString, MqttString};
+///
+/// // From &str
+/// let s1 = "test".into_mqtt_string().unwrap();
+/// assert_eq!(s1.as_str(), "test");
+///
+/// // From String
+/// let s2 = String::from("test").into_mqtt_string().unwrap();
+/// assert_eq!(s2.as_str(), "test");
+///
+/// // From MqttString (no additional allocation)
+/// let mqtt_str = MqttString::new("test").unwrap();
+/// let s3 = mqtt_str.into_mqtt_string().unwrap();
+/// assert_eq!(s3.as_str(), "test");
+/// ```
+pub trait IntoMqttString {
+    /// Convert self into an MqttString
+    fn into_mqtt_string(self) -> Result<MqttString, MqttError>;
+}
+
+impl IntoMqttString for &str {
+    fn into_mqtt_string(self) -> Result<MqttString, MqttError> {
+        MqttString::new(self)
+    }
+}
```

Comment:
If you don't mind me asking, why create a custom trait instead of using `TryInto`? It looks like there are already some `TryInto` implementations, so this using it would duplicating them. And there's already a blanket implementation for types `TryInto` themselves. I implemented it here: https://github.com/smaeul/mqtt-protocol-core/commit/800b960f886528da6e2ae936875af678fb68acf2